src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,153 @@
+/* eslint-disable @typescript-eslint/indent */
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [loader, setLoader] = useState(false);
+  const [error, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort');
+  const reversed = searchParams.get('reversed');
+  const query = searchParams.get('query');
+  const sex = searchParams.get('sex');
+  const century = searchParams.getAll('centuries') || [];
+
+  useEffect(() => {
+    setLoader(true);
+    getPeople()
+      .then(data => {
+        if (data) {
+          setPeoples(data);
+        }
+      })
+      .catch(() => {
+        setError(true);
+      })
+      .finally(() => {
+        setLoader(false);
+      });
+  }, []);
+
+  const getSortedPeople = useCallback(() => {
+    let sortedPeople = [...peoples];
+
+    if (sortedPeople && sort) {
+      switch (sort) {
+        case 'name':
+          sortedPeople.sort((a, b) => a.name.localeCompare(b.name));
+          break;
+        case 'sex':
+          sortedPeople.sort((a, b) => a.sex.localeCompare(b.sex));
+          break;
+        case 'born':
+          sortedPeople.sort((a, b) => a.born - b.born);
+          break;
+        case 'died':
+          sortedPeople.sort((a, b) => a.died - b.died);
+          break;
```

Comment:
```suggestion
        case 'born':
        case 'died':
          sortedPeople.sort((a, b) => a[sort]- b[sort]);
          break;
```
we can combine some cases